core/src/test/java/org/hisp/dhis/android/core/event/EventCallShould.java

```diff
@@ -0,0 +1,166 @@
+package org.hisp.dhis.android.core.event;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.hisp.dhis.android.core.calls.Call.MAX_UIDS;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.hisp.dhis.android.core.calls.Call;
+import org.hisp.dhis.android.core.common.Payload;
+import org.hisp.dhis.android.core.data.api.FieldsConverterFactory;
+import org.hisp.dhis.android.core.data.api.FilterConverterFactory;
+import org.hisp.dhis.android.core.data.database.DatabaseAdapter;
+import org.hisp.dhis.android.core.data.file.ResourcesFileReader;
+import org.hisp.dhis.android.core.data.server.Dhis2MockServer;
+import org.hisp.dhis.android.core.resource.ResourceHandler;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+import okhttp3.mockwebserver.RecordedRequest;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+public class EventCallShould {
+    private Call<Response<Payload<Event>>> eventCall;
+
+    @Mock
+    private EventService eventService;
+
+    @Mock
+    private DatabaseAdapter databaseAdapter;
+
+    @Mock
+    private ResourceHandler resourceHandler;
+
+    @Mock
+    private EventHandler eventHandler;
+
+    @Mock
+    private Date serverDate;
+
+    Dhis2MockServer dhis2MockServer;
+    Retrofit retrofit;
+
+    @Before
+    @SuppressWarnings("unchecked")
+    public void setUp() throws IOException {
+        dhis2MockServer = new Dhis2MockServer(new ResourcesFileReader());
+
+        //TODO reuse dhis2MockServer from AndroidTest
+
+        retrofit = new Retrofit.Builder()
+                .baseUrl(dhis2MockServer.getBaseEndpoint())
+                .addConverterFactory(JacksonConverterFactory.create(new ObjectMapper()))
+                .addConverterFactory(FilterConverterFactory.create())
+                .addConverterFactory(FieldsConverterFactory.create())
+                .build();
+
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        dhis2MockServer.shutdown();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void throw_illegal_argument_exception_if_uids_size_exceeds_the_limit() {
+        EventCall eventCall = givenAEventCallByUIds(MAX_UIDS + 1);
+    }
+
+    @Test
+    public void create_event_call_if_uIds_size_does_not_exceeds_the_limit() {
+        EventCall eventCall = givenAEventCallByUIds(MAX_UIDS);
```

Comment:
@xurxodev don't we miss an assertion here? It looks like we're not checking the eventCall is created, right?